scripts/irods/test/test_all_rules.py

```diff
@@ -663,6 +663,82 @@ def test_msiServerMonPerf_default_3736(self):
 
         self.rods_session.assert_icommand("irule -F " + rule_file);
 
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for irods_rule_language REP')
+    def test_msiAddKeyValToMspStr_works_with_empty_string_6918(self):
+        collection_to_free = os.path.join(self.admin.session_collection, 'freeMe')
+
+        # Generate trash
+        self.admin.assert_icommand(['itouch', collection_to_free])
+        self.admin.assert_icommand(['irm', collection_to_free])
+
+        rule_file="test_msiAddKeyValToMspStr.r"
+        rule_string= '''
+msiTestAddKeyValToMspStr {{
+    *Flag = "";
+    *Coll = "{trash_col}";
+    msiAddKeyValToMspStr("irodsAdminRmTrash", "", *Flag);
+    msiRmColl(*Coll, *Flag, *result);
+}}
+
+INPUT null
+OUTPUT ruleExecOut
+'''.format(trash_col=os.path.join(self.admin.session_collection_trash))
+
+        with open(rule_file, 'w') as f:
+            f.write(rule_string)
+
+        rep_name = 'irods_rule_engine_plugin-irods_rule_language-instance'
+        self.admin.assert_icommand(['irule', '-r', rep_name, '-F', rule_file])
+
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for irods_rule_language REP')
+    def test_msiRmColl_removes_collection_6918(self):
+        collection_to_free = os.path.join(self.admin.session_collection, 'freeMe')
+
+        self.admin.assert_icommand(['imkdir', collection_to_free])
+
+        rule_file="test_msiRmCollSimple.r"
+        rule_string= '''
+msiTestAddKeyValToMspStr {{
```

Comment:
Let's update the name of the rule to reflect the test

Reply:
Should be done!